ncclient/operations/third_party/iosxr/rpc.py

```diff
@@ -0,0 +1,13 @@
+from ncclient.xml_ import *
+
+from ncclient.operations.rpc import RPC
+from ncclient.operations.rpc import RPCReply
+from ncclient.operations.rpc import RPCError
+from ncclient import NCClientError
+
+class ExecuteRpc(RPC):
```

Comment:
Can you please explain what this class is doing and why? I think I can see what you are trying to achieve, but please be explicit!

Reply:
Currently the default device handler ends up using Junos Device handler, due to which common RPCs like commit ends up failing on Cisco IOS-XR devices as example Commit rpc in juniper is <commit-configuration/> however in Cisco it is <commit/>, currently to workaround this, I have just mirrored the juniper ExecuteRPC class and added it in Cisco IOS-XR handler to allow us to proceed with the testing.

Reply:
I have been using the default device handler successfully for RPCs such as commit with IOS XR for a long time, so I’m a little confused. What problems have you been experiencing?

Reply:
When I use default handler it doesnt generate the right commit rpc 
example below 
<?xml version="1.0" encoding="UTF-8"?><nc:rpc xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:18792888-9b0a-4847-a098-1be601c479e1"><commit-configuration/></nc:rpc>

Reply:
Just to add - custom RPC does work with default handler

Reply:
> When I use default handler it doesnt generate the right commit rpc
> example below:
> 
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> <nc:rpc xmlns:nc="urn:ietf:params:xml:ns:netconf:base:1.0" message-id="urn:uuid:18792888-9b0a-4847-a098-1be601c479e1">
>   <commit-configuration/>
> </nc:rpc>
> ```

This example is from the JunOS handler, AFAIK. As I said, I've been using the default device handler, and others the IOS XR handler, for a number of years, including commit operations.

Reply:
Can you please provide a small test harness that demonstrates the problem you have encountered? There should be no code path that allows the JunOS device handler to provide operations for the default device handler.